web/src/app/util/ISOPickers.js

```diff
@@ -0,0 +1,139 @@
+import React, { useState, useEffect } from 'react'
+import { DatePicker, TimePicker, DateTimePicker } from '@material-ui/pickers'
+import { useSelector } from 'react-redux'
+import { urlParamSelector } from '../selectors'
+import { DateTime } from 'luxon'
+import { TextField, InputAdornment, IconButton } from '@material-ui/core'
+
+import Modernizr from '../../modernizr.config'
+import { DateRange } from '@material-ui/icons'
+
+function hasInputSupport(name) {
+  if (new URLSearchParams(location.search).get('nativeInput') === '0') {
+    return false
+  }
+
+  return Modernizr.inputtypes[name]
+}
+
+function useISOPicker(
+  { value, onChange, timeZone, ...otherProps },
+  { format, truncateTo, type, Fallback },
+) {
+  const native = hasInputSupport(type)
+  const params = useSelector(urlParamSelector)
+  const zone = timeZone || params('tz', 'local')
+  const dtValue = DateTime.fromISO(value, { zone })
+  const [inputValue, setInputValue] = useState(dtValue.toFormat(format))
+
+  const parseInput = input => {
+    if (input instanceof DateTime) return input
+    if (!input) return null
+
+    const dt = DateTime.fromFormat(input, format, { zone })
+    if (dt.isValid) {
+      if (type === 'time') {
+        return dtValue.set({
+          hour: dt.hour,
+          minute: dt.minute,
+        })
+      }
+      return dt
+    }
+
+    const iso = DateTime.fromISO(input)
+    if (iso.isValid) return iso
+
+    return null
+  }
+  const isoInput = input => {
+    const val = parseInput(input)
+    return val
+      ? val
+          .startOf(truncateTo)
+          .toUTC()
+          .toISO()
+      : ''
+  }
+
+  useEffect(() => {
+    setInputValue(dtValue.toFormat(format))
+  }, [value, zone])
+
+  const handleChange = e => {
+    setInputValue(e.target.value)
+
+    const newVal = isoInput(e.target.value)
+    if (newVal && newVal !== value) {
+      onChange(newVal)
```

Comment:
Can you add a note here as to what `onChange`'s intended use is? Hard to tell with it being passed through as a param, especially with `handleChange` presumably being that